packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,380 @@
+  ///
+  /// If this property is null, then the side will be width 1.
+  final BorderSide? side;
+
+  /// The shape of the checkbox.
+  ///
+  /// If this property is null then the shape will be a [RoundedRectangleBorder]
+  /// with a circular corner radius of 4.0.
+  final OutlinedBorder? shape;
+
+  /// The width of a checkbox widget.
+  static const double width = 18.0;
+
+  @override
+  State<CupertinoCheckbox> createState() => _CupertinoCheckboxState();
+}
+
+class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProviderStateMixin, ToggleableStateMixin {
+  final _CheckboxPainter _painter = _CheckboxPainter();
+  bool? _previousValue;
+
+  bool focused = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _previousValue = widget.value;
+  }
+
+  @override
+  void didUpdateWidget(CupertinoCheckbox oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.value != widget.value) {
+      _previousValue = oldWidget.value;
+    }
+  }
+
+  @override
+  void dispose() {
+    _painter.dispose();
+    super.dispose();
+  }
+
+  @override
+  ValueChanged<bool?>? get onChanged => widget.onChanged;
+
+  @override
+  bool get tristate => widget.tristate;
+
+  @override
+  bool? get value => widget.value;
+
+  void onFocusChange(bool value) {
+    if (focused != value) {
+      focused = value;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    final Color effectiveActiveColor = widget.activeColor
+      ?? CupertinoColors.activeBlue;
+    final Color? inactiveColor = widget.inactiveColor;
+    final Color effectiveInactiveColor = inactiveColor
+      ?? CupertinoColors.inactiveGray;
+
+    final Color effectiveFocusOverlayColor = widget.focusColor
+      ?? HSLColor
+          .fromColor(effectiveActiveColor.withOpacity(0.80))
+          .withLightness(0.69).withSaturation(0.835)
+          .toColor();
+
+    final Color effectiveCheckColor = widget.checkColor
+      ?? CupertinoColors.white;
+
+    return Semantics(
+      checked: widget.value ?? false,
+      mixed: widget.tristate ? widget.value == null : null,
+      child: buildToggleable(
+        focusNode: widget.focusNode,
+        autofocus: widget.autofocus,
+        onFocusChange: onFocusChange,
+        size: const Size(48.0, 48.0),
+        painter: _painter
+          ..focusColor = effectiveFocusOverlayColor
+          ..isFocused = focused
+          ..downPosition = downPosition
+          ..activeColor = effectiveActiveColor
+          ..inactiveColor = effectiveInactiveColor
+          ..checkColor = effectiveCheckColor
+          ..value = value
+          ..previousValue = _previousValue
+          ..isActive = widget.onChanged != null
+          ..shape = widget.shape ?? RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(4.0),
+          )
+          ..side = widget.side,
+      ),
+    );
+  }
+}
+
+const double _kEdgeSize = CupertinoCheckbox.width;
```

Comment:
Why can't the use locations for this use `CupertinoCheckbox.width` directly?  It is also a const, and would show where it comes from.